packages/compiler/test/checker/templates.ts

```diff
@@ -179,14 +179,15 @@ describe("compiler: templates", () => {
     testHost.addCadlFile(
       "main.cadl",
       `
-        @test model A<T, X = T> { a: T, b: X }
-        model B { 
+        model A<T, X = T> { a: T, b: X }
+        @test model B { 
```

Comment:
Hmm, is this change necessary? I feel like the previous way should work as well...

Reply:
Yeah I think it still works, I can revert.
As I was trying to solve this problem I got this test failing and thought it was because of this. I am a bit confused on why it works. Wouldn't the model be returned be the uninstantiated template?
Or this just work because there is a single instantiation of A and the @test decorator just act this way?

Reply:
It's the latter. `@test` is basically last-wins for each type name. We run decorators when the model gets instantiated, and since there is only one instantiation, A will happen to be what you want to test in this case. I kind of like this test approach because it's sensitive to changes in decorator+instantiation logic which at least previously was an area of buggyness.

Reply:
make sense, are you saying you like the way the test was defined before?

Reply:
yeah I like the way it was before, but considering that we probably have pretty good coverage for decorators+templated types elsewhere I would be fine keeping the change. It does add more verbosity though!